### ISSUE TITLE
[Feature] Implement Monero Restore Wallet URI Scheme (Gift Card Redeeming)

### ISSUE THREAD

**Issue by CryptoGrampy** [cake-tech/cake_wallet#196]
Getting gift cards into the Monero ecosystem is a sort of chicken and egg problem.  We already have an agreed-upon data specification for encoding a wallet into a QR code (like a gift card), but we don't have the ability to create a wallet QR, AND we also don't have the ability to redeem a QR code wallet.  We're never going to get gift cards until one side budges.  

I would argue that implementing the redeeming end of the gift card ecosystem is probably the most important as it needs to be trusted, established, an easy on-ramp for new users, ability to scan a QR, and be cross-platform.  Cake-Wallet would perfectly fit this bill as it covers iOS and Android and has a built in camera.  

Ideally, a user should be able to scan a QR containing the monero_restore wallet URI, they hit the browser and Cake Wallet should open up with a redeem gift card/restore wallet UI.  Additionally, if one opens Cake Wallet, 'Redeem from gift card' should be an option for creating a new wallet.  Users should be encouraged to create their own personal wallet and send gift card funds there (could also be part of the UI process.... would you like to create a new wallet and send funds to this wallet?) etc.

Once this ability is in place, it will encourage developers to build gift card generators and begin dispersing.  

https://github.com/monero-project/monero/wiki/URI-Formatting#wallet-definition-scheme
https://bounties.monero.social/posts/27/gift-wallet-on-paper

**Comment by CryptoGrampy** [comment]
Obviously, in addition to the above, it would be amazing if Cake had a 'create gift card' wizard that could be accessed from a wallet that ended with a QR with cake app download instructions that could be printed or opened with an email app on the device.

**Comment by SamsungGalaxyPlayer** [comment]
Related to https://github.com/monero-project/meta/issues/729
